packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -380,6 +380,8 @@ export class ProjectLifecycleManager {
   }
 
   private _setCurrentProject (projectRoot: string) {
+    process.chdir(projectRoot)
```

Comment:
I also think this is a bad pattern, we chdir in several places because many code paths assume we are in the project dir. I think it would be cleaner if we just passed `projectRoot` where it's needed. But I'm following tradition here.